emissary-core/src/destination/session/mod.rs

```diff
@@ -656,7 +664,42 @@ impl<R: Runtime> SessionManager<R> {
                     SessionError::Malformed
                 })?;
 
-                // TODO: verify `DateTime`
+                // locate `DateTime` message from the clove set
+                let Some(GarlicMessageBlock::DateTime { timestamp }) = clove_set
```

Comment:
You could go over the cloves only once with `filter_map().collect()` and then ensure that there is exactly one entry in the vector and if not, log appropriate warning